core/src/main/java/bio/terra/pearl/core/service/consent/ConsentTaskDispatcher.java

```diff
@@ -98,12 +103,14 @@ public static boolean checkIsEnrolleeConsented(List<ParticipantTask> participant
 
     /** builds the consent tasks, does not add them to the event or persist them */
     public List<ParticipantTask> buildTasks(Enrollee enrollee,
-                                                   EnrolleeRuleData enrolleeRuleData,
+                                            EnrolleeRuleData enrolleeRuleData,
                                                    UUID portalParticipantUserId,
                                                    List<StudyEnvironmentConsent> studyEnvConsents) {
         List<ParticipantTask> tasks = new ArrayList<>();
         for (StudyEnvironmentConsent studyConsent : studyEnvConsents) {
-            if (EnrolleeRuleEvaluator.evaluateRule(studyConsent.getEligibilityRule(), enrolleeRuleData)) {
+            if (enrollee.isSubject() && enrolleeSearchExpressionParser
```

Comment:
I think the logic we want is  "if rule is blank, apply default rule, otherwise parse rule."  (See `evaluateRuleChecked` in EnrolleeRuleEvaluator).  Otherwise we've foreclosed the ability to have forms that apply to non-subjects.

Reply:
Yeah, I agree with you that it shouldn't be hardcoded, but I want to sync with Erin on how these rules should work regarding isSubject. I am not sure how I feel about it being a default rule - let's say you wanted to give the survey to both enrollees and their proxies. Then, you have to do `{enrollee.subject} = true and {enrollee.subject} != true`? That feels weird/arcane to me.

Thinking about it, it feels like these subject checks should exist outside of search expressions, at least in the context of survey eligibility rules. It feels like, to me, there should be two checkboxes above the eligibility rule, `Give survey to enrollees` and `Give survey to proxies` (in the case that the study has proxies), and you can either select one or both. Then there isn't any magic happening in the backend that you have to work around and it's super clear who is and is not being given this survey.

Reply:
Also, I made the assumption that consent surveys would only be for enrollees, because they're the ones consenting to the survey, but maybe this is not always true.